### PR TITLE
Build ceph from source and from our fork

### PR DESCRIPTION
**Pull request by bassamtabbara**

### CODE REVIEW

File: sys-cluster/ceph/ceph-9999.ebuild [hunk 1]
```diff
@@ -4,17 +4,16 @@
...
-
-if [[ ${PV} == *9999* ]]; then
-	inherit git-r3
-	EGIT_REPO_URI="
-		git://github.com/ceph/ceph.git
-		https://github.com/ceph/ceph.git"
-	SRC_URI=""
+inherit autotools eutils systemd versionator flag-o-matic cros-workon
+
+CROS_WORKON_PROJECT="quantum/ceph"
```
Comment (jbw976) [inline sys-cluster/ceph/ceph-9999.ebuild]: did we just now fork from ceph to create our own ceph source repo?  if so, do we have a good story (automated scripts) for merging from upstream yet?  curious.

File: sys-cluster/ceph/ceph-9999.ebuild [hunk 2]
```diff
@@ -40,16 +39,11 @@ COMMON_DEPEND="
 	sys-fs/xfsprogs
 	"
 DEPEND="${COMMON_DEPEND}
-"
+	"
```
Comment (jbw976) [inline sys-cluster/ceph/ceph-9999.ebuild]: is this whitespace change important?

File: eclass/cros-workon.eclass [hunk 3]
```diff
@@ -110,6 +110,11 @@ local_clone() {
 		rm -rf "${S}" || die
 		return 1
 	fi
+
+	# initial any git submodules. We use --reference here to reduce the
+	# amount of objects fetched over the network. There is no --shared
+	# option for git submodules. This is a no-op when there are no submodules
+	git -C "${S}" submodule update --init --recursive --reference "${path}"
```
Comment (jbw976) [inline eclass/cros-workon.eclass]: could this submodule init logic be easily pushed upstream? or is it not relevant there?

**Comment by jbw976** [comment]
LGTM

File: eclass/cros-workon.eclass (hunk 3 above)
Comment (bassamtabbara) [inline eclass/cros-workon.eclass]: I think so. its only needed when building from git repos that use submodules.

File: sys-cluster/ceph/ceph-9999.ebuild (hunk 2 above)
Comment (bassamtabbara) [inline sys-cluster/ceph/ceph-9999.ebuild]: consistency with RDEPEND later :-)

**Pull request merged**